crates/turbopack-css/src/module_asset.rs

```diff
@@ -193,21 +197,37 @@ impl EcmascriptChunkItem for ModuleChunkItem {
                 }
                 code += "});\n";
                 EcmascriptChunkItemContent {
-                    inner_code: code.into(),
-                    // TODO: We generate a minimal map for runtime code so that the filename is
+                    inner_code: code.clone().into(),
+                    // We generate a minimal map for runtime code so that the filename is
                     // displayed in dev tools.
+                    source_map: Some(generate_minimal_source_map(
+                        format!("{}.js", self.module.path().await?.path),
+                        code,
+                    )),
                     ..Default::default()
                 }
             }
             ParseResult::NotFound | ParseResult::Unparseable => {
+                let code = "__turbopack_export_value__({});\n";
                 EcmascriptChunkItemContent {
-                    inner_code: "__turbopack_export_value__({});\n".into(),
-                    // TODO: We generate a minimal map for runtime code so that the filename is
+                    inner_code: code.into(),
+                    // We generate a minimal map for runtime code so that the filename is
                     // displayed in dev tools.
+                    source_map: Some(generate_minimal_source_map(
+                        format!("{}.js", self.module.path().await?.path),
+                        code.into(),
+                    )),
                     ..Default::default()
                 }
             }
         }
         .cell())
     }
 }
+
+fn generate_minimal_source_map(filename: String, source: String) -> ParseResultSourceMapVc {
+    let sm: Arc<SourceMap> = Default::default();
+    sm.new_source_file(FileName::Custom(filename), source);
+    let map = ParseResultSourceMap::new(sm, vec![(BytePos(1), LineCol { line: 0, col: 0 })]);
```

Comment:
For breakpoints and stepping to work correctly you at least need to generate a SourceMapping per line.

Reply:
I changed this to use parse and emit to generate the source map, which made the source map not minimal and have more cost for performance, or do you suggest create the mappings with traversing the `source` line by line? It seems inconvenient but more performance.

Reply:
I think his suggestion was to add multiple items to this `vec` to represent each line of the output source file. If your interested in undo the changes, this should be as simple as a `source.lines()` and build up all the lines.

Reply:
I will undoing the changes and have a try later when I have free time.